pkg/common/utils.go

```diff
@@ -224,6 +227,9 @@ func ZeroTimeInUnixNano(t time.Time) *int64 {
 }
 
 func WaitForCondition(eval func() bool, interval time.Duration, timeout time.Duration) error {
+	if interval > timeout {
```

Comment:
Personally, changing the `interval` is a bit awkward since that breaks the definition. Maybe we can rewrite this method.

```go
	deadline := time.Now().Add(timeout)
	for {
		if eval() {
			return nil
		}

		if time.Now().After(deadline) || time.Now().Add(interval).After(deadline) {
			return fmt.Errorf("timeout waiting for condition")
		}

		time.Sleep(interval)
	}
```

In short, we return timeout error early if it will be timeout due to sleeping.

Reply:
Can we define a fixed error in `common.errors.go` (new file) like this:
```
package common

import (
	"errors"
)

var (
	// ErrorTimeout returned if waiting for a condition times out
	ErrorTimeout = errors.New("timeout waiting for condition")
)
```
That will make error chaining easy and allow using `errors.Is()` in all kinds of places. It would be inline with the constants definition we already have